python/dialects/aiex.py

```diff
@@ -836,3 +836,64 @@ def dma_start_bd_chain_for(symbol, args, alloc, *pyargs, **kwargs):
     return DMAStartBdChainForOp(
         T.index(), chain_sym, args, alloc_sym, *pyargs, **kwargs
     )
+
+
+def shim_dma_bd(
+    mem,
+    offset: int | None = None,
+    sizes: MixedValues | None = None,
+    strides: MixedValues | None = None,
+    transfer_len: int | None = None,
+):
+    if offset is None:
+        offset = 0
+    if sizes is None:
+        sizes = [0] * 4
+    if strides is None:
+        strides = [0] * 3 + [1]
+
+    if transfer_len is None:
+        if len(sizes) >= 4:
+            # For shim dma bd, highest dimension is repeat count which is not included in the length
+            transfer_len = np.prod(sizes[1:])
+        else:
+            # If does not have highest dimension, then we can take the product of all dimensions
+            transfer_len = np.prod(sizes)
+    dimensions = list(zip(sizes, strides))
+    dma_bd(mem, offset=offset, len=transfer_len, dimensions=dimensions)
+
+
+_orig_dma_await_task = dma_await_task
+
+
+def dma_await_task(*args: DMAConfigureTaskForOp):
+    if len(args) == 0:
+        raise ValueError(
+            "dma_await_task must receive at least one DMAConfigureTaskForOp to wait for"
+        )
+    for dma_task in args:
+        _orig_dma_await_task(dma_task)
+
+
+_orig_dma_free_task = dma_free_task
+
+
+def dma_free_task(*args: DMAConfigureTaskForOp):
+    if len(args) == 0:
+        raise ValueError(
+            "dma_free_task must receive at least one DMAConfigureTaskForOp to free"
+        )
+    for dma_task in args:
+        _orig_dma_free_task(dma_task)
+
+
+_orig_dma_start_task = dma_start_task
+
+
+def dma_start_task(*args: DMAConfigureTaskForOp):
+    if len(args) == 0:
+        raise ValueError(
+            "dma_start_task must receive at least one DMAConfigureTaskForOp to free"
+        )
+    for dma_task in args:
+        _orig_dma_start_task(dma_task)
```

Comment:
There is a tradition of not forcing a user to make a list for a single element elsewhere in the python bindings (e.g., here: https://github.com/Xilinx/mlir-aie/blob/774e0e6ae680441a6ea54638901836a8a4b5d69b/python/dialects/aie.py#L441). I believe the implementation I chose is not an uncommon way to implement this pattern in python. I've also used this syntax here: https://github.com/Xilinx/mlir-aie/blob/774e0e6ae680441a6ea54638901836a8a4b5d69b/python/dialects/aiex.py#L38

Since I hadn't made other functions plural (since they work for single elements) I did not make this function plural for consistency.

I don't think adding attributes is a problem as you can easily add keyword arguments to this function still.

I personally think the current implementation is ok, but if you feel strongly I can change the implementation and naming convention of this and other functions to be consistent in another way.